code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -42,18 +42,18 @@ GLOBAL_LIST_EMPTY(antagonists)
 
 /datum/antagonist/Destroy(force, ...)
 	qdel(objective_holder)
-	remove_owner_from_gamemode()
 	GLOB.antagonists -= src
-	if(!silent)
-		farewell()
-	remove_innate_effects()
-	antag_memory = null
-	var/datum/team/team = get_team()
-	team?.remove_member(owner)
 	if(owner)
```

Comment:
Code looks fine, but this may be better as an early return?, i.e.
```
if(!owner)
    return ..()
rest of the code dependent on owner
```

Reply:
We can, but it will look like:
```
if(!owner)
    return ..()
rest of the code dependent on owner
return ..()
```

Reply:
Or I can just move all this to separate `proc/remove_owner_dependencies()`